Arriba/Arriba.Client/Model/SecureDatabase.cs

```diff
@@ -51,6 +53,142 @@ public SecurityPermissions Security(string tableName)
             }
         }
 
+        public override T Query<T>(IQuery<T> query)
+        {
+            throw new ArribaException("Use Query overload which takes isCurrentUserIn on SecureDatabase.");
+        }
+
+        public T Query<T>(IQuery<T> query, Func<SecurityIdentity, bool> isCurrentUserIn)
+        {
+            ExecutionDetails preExecuteDetails = new ExecutionDetails();
+            query = SecureQuery(query, isCurrentUserIn, preExecuteDetails);
+
+            T result = base.Query<T>(query);
+            if(result is IBaseResult)
+            {
+                ((IBaseResult)result).Details.Merge(preExecuteDetails);
+            }
+
+            return result;
+        }
+
+        public IQuery<T> SecureQuery<T>(IQuery<T> query, Func<SecurityIdentity, bool> isCurrentUserIn, ExecutionDetails details)
```

Comment:
The return value is weird here too.  It can't return a clone of IQuery so it's just returning the same thing you pass in. Probably doesn't need to return here.  Or return the ExecutionDetails.